tests/foreman/ui/test_subscription.py

```diff
@@ -290,30 +289,28 @@ def test_positive_view_vdc_subscription_products(session):
     repos_collection.setup_content(
         org.id, lce.id, upload_manifest=True, rh_subscriptions=[DEFAULT_SUBSCRIPTION_NAME]
     )
-    with VirtualMachine() as vm:
-        setup_virtual_machine(
-            vm,
-            org.label,
-            activation_key=repos_collection.setup_content_data['activation_key']['name'],
-            install_katello_agent=False,
-        )
-        with session:
-            session.organization.select(org.name)
-            session.contenthost.add_subscription(vm.hostname, VDC_SUBSCRIPTION_NAME)
-            provided_products = session.subscription.provided_products(VDC_SUBSCRIPTION_NAME)
-            # ensure that subscription provided products list is not empty and that the product is
-            # in the provided products.
-            assert provided_products and product_name in provided_products
-            content_products = session.subscription.content_products(VDC_SUBSCRIPTION_NAME)
-            # ensure that subscription enabled products list is not empty and that product is in
-            # content products.
-            assert content_products and product_name in content_products
+    setup_virtual_machine(
+        rhel7_contenthost,
+        org.label,
+        activation_key=repos_collection.setup_content_data['activation_key']['name'],
+        install_katello_agent=False,
+    )
+    with session:
+        session.organization.select(org.name)
+        session.contenthost.add_subscription(rhel7_contenthost.hostname, VDC_SUBSCRIPTION_NAME)
+        provided_products = session.subscription.provided_products(VDC_SUBSCRIPTION_NAME)
+        # ensure that subscription provided products list is not empty and that the product is
+        # in the provided products.
+        assert provided_products and product_name in provided_products
```

Comment:
```suggestion
        assert product_name in provided_products
```
As long as provided_products is always a list (or other iterable), then the first check isn't needed.
This type of check is used throughout the other tests in this PR.

I know this was part of the original tests, but could you clean these up as well? :)

Reply:
Done. :)